### DISCUSSION TITLE
`create-next-app` should use npm by default when called with npx

### DISCUSSION THREAD

**Discussion by Nsttt** [vercel/next.js#41090] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.16.0
      npm: 8.11.0
      Yarn: 1.22.19
      pnpm: 7.11.0
    Relevant packages:
      next: 12.3.2-canary.16
      eslint-config-next: 12.3.1
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

Currently it's not possible to use `npx create-next app` or `npm create next-app` using npm as the default package manager without passing the `--use-npm` flag, if you create it without passing it, the CLI will default to install its dependencies using `yarn`. 

This seems unnecessary since we can check whether the user calls the CLI through `npm`. 

Other CLI tools such as [create-t3-app](https://github.com/t3-oss/create-t3-app/blob/main/cli/src/utils/getUserPkgManager.ts) already does this, and I think it should be implemented into `create-next-app`.

### Expected Behavior

The CLI should be deterministic by which package manager has it been called with. If I create an app using `npx` my depencies should be installed using `npm`.

This already happens when you create it using `pnpm create`.

### Link to reproduction

Create a new app using `npx create-next-app` or `npm create next-app``

### To Reproduce

1. Create a new app using `npx create-next-app` or `npm create next-app`
2. The app will default to `yarn` to install its packages.

**Comment by balazsorban44** [comment]
Thanks for raising this. The current behavior is unless a `--use-npm` or `--use-pnpm` flag is passed, we gracefully parse your environment to find the package manager of choice as shown:

https://github.com/vercel/next.js/blob/b7fa280faaa3391073acc5847eac7f245d18aa7b/packages/create-next-app/index.ts#L129

https://github.com/vercel/next.js/blob/canary/packages/create-next-app/helpers/get-pkg-manager.ts

you could argue we should use the one based on the manager you call the CLI with, though it might be a breaking change. Are you interested in opening a PR for this?

**Reply by Nsttt** [reply]
Sure, will love to. I'll look into it the following week :)

**Comment by Nsttt** [comment]
Made a very basic change that seems to work.  #41279 

There may be more things that needs to be done such as changing the docs or anything else, hit me up if that's the case and I'll make the changes :)
